cpp/src/arrow/compute/kernels/temporal_internal.h

```diff
@@ -105,6 +105,39 @@ struct NonZonedLocalizer {
     return t;
   }
 
+  template <typename Duration, typename Unit>
+  Duration FloorTimePoint(int64_t t, const RoundTemporalOptions* options) const {
```

Comment:
Nit, but why not pass the options by const-reference? I don't think it's allowed to omit them.

Reply:
Done.